ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/ErrorFactories.scala

```diff
@@ -496,6 +488,32 @@ class ErrorFactories private (errorCodesVersionSwitcher: ErrorCodesVersionSwitch
     grpcError(statusBuilder.build())
   }
 
+  /** Transforms Protobuf [[Status]] objects, possibly including metadata packed as [[ErrorInfo]] objects,
+    * into exceptions with metadata in the trailers.
+    *
+    * Asynchronous errors, i.e. failed completions, contain Protobuf [[Status]] objects themselves.
+    *
+    * NOTE: The length of the Status message is truncated to a reasonable size for satisfying
+    *        the Netty header size limit - as the message is also incorporated in the header, bundled in the gRPC metadata.
+    * @param status A Protobuf [[Status]] object.
+    * @return An exception without a stack trace.
+    */
+  def grpcError(status: Status): StatusRuntimeException = {
+    val newStatus =
+      Status
+        .newBuilder(status)
+        .setMessage(truncated(status.getMessage))
+    new NoStackTraceApiException(
+      StatusProto.toStatusRuntimeException(newStatus.build)
+    )
+  }
+
+  private def truncated(message: String): String = {
+    val maxMessageLength =
```

Comment:
Can you make this arbitrary limit more explicit, please? E.g., by moving its definition to above `grpcError` and referencing it from the latter method's ScalaDoc. The same should be done for `internalAuthenticationError` to make this logic transparent.